utilities/xmlrpcpp/src/XmlRpcClient.cpp

```diff
@@ -458,15 +459,25 @@ XmlRpcClient::parseResponse(XmlRpcValue& result)
     if ( ! result.fromXml(_response, &offset)) {
       XmlRpcUtil::error("Error in XmlRpcClient::parseResponse: Invalid response value. Response:\n%s", _response.c_str());
       _response = "";
+      _header = "";
       return false;
     }
   } else {
     XmlRpcUtil::error("Error in XmlRpcClient::parseResponse: Invalid response - no param or fault tag. Response:\n%s", _response.c_str());
     _response = "";
+    _header = "";
     return false;
   }
       
   _response = "";
+
+  // Close connection if protocol is HTTP/1.0
+  if (_header.size() > 8 && _header[5] == '1' && _header[7] == '0') {
```

Comment:
Why not continue to check for the full string? Otherwise this would match e.g. `OTHER120`.

Also why `> 8`? Isn't the header size exactly 8 for `HTTP/1.0`?